lib/trino-hdfs/src/main/java/io/trino/hdfs/TrinoFileSystemCache.java

```diff
@@ -399,6 +401,23 @@ public RemoteIterator<LocatedFileStatus> listFiles(Path path, boolean recursive)
         {
             return fs.listFiles(path, recursive);
         }
+
+        @Override
+        public void batchDeleteFiles(Collection<Path> paths)
+                throws IOException
+        {
+            FileSystem rawFileSystem = getRawFileSystem();
+            if (rawFileSystem instanceof FileSystemWithBatchDelete fileSystemWithBatchDelete) {
+                fileSystemWithBatchDelete.batchDeleteFiles(paths);
+            }
+            else {
+                for (Path path : paths) {
+                    if (!fs.delete(path, true) && fs.exists(path)) {
```

Comment:
why `&& fs.exists(path)`? worth explaining?

Reply:
That's a mistake I copied deleteDirectory instead of deleteFile code..